### DISCUSSION TITLE
How to get IPF X instead of the out of plane (IPF Z)

### DISCUSSION THREAD

**Discussion by sarialzerey** [mtex-toolbox/mtex#2041] [Ask Anything] [answered]
Hi, 

I used the below code to change IPF Z to IPF X but when I displayed the crystal shape, it seemed the same for both IPF Z and IPF X as shown in the images.

>>> I'm not sure if this is the right code to get IPF X. The grains look the same in both cases and this doesn't seem right <<<

![ipfz](https://github.com/mtex-toolbox/mtex/assets/141929646/0cf75c93-b671-462b-989a-60430ee38834)
![ipfx](https://github.com/mtex-toolbox/mtex/assets/141929646/a3f2c8d3-c535-4313-8a5b-adb30bbb3bb3)

%Changing IPF direction from out of plane Z to X

%create a color map of the IPF orientations of the selected material
oM = ipfHSVKey(ebsd('copper'));
%define the direction of the ipf
oM.inversePoleFigureDirection = xvector;
%convert the ebsd map orientations to a color based on the IPF
color = oM.orientation2color(ebsd('copper').orientations);
figure; plot(ebsd('copper'),color);

%% show the crystal shape on the grains
cS = crystalShape.cube(ebsd('copper').CS)
hold on
largeGrains = grains(grains.grainSize>10)%select grains by size
CrystalShapeOrientation=plot(largeGrains('copper'), 0.35*cS, 'FaceColor', 'none') %scaling
set(get(get(CrystalShapeOrientation,'Annotation'),'LegendInformation'),'IconDisplayStyle','off')

**Comment by zmichels** [comment, marked as answer]
IPF maps are a coloring technique, and the color is changing as you change the relative direction definition -- so everything appears to be working with your code. But the crystal shapes are displaying the actual crystal lattice orientations... which are not being rotated anywhere in your code... so they look the same in both plots.

**Reply by sarialzerey** [reply]
Thanks for the response, Zmichels.
So you mean it's not possible to change the crystal orientation?

**Reply by zmichels** [reply]
It is possible to rotate your orientations. You can either rotate the entire dataset... which will also relocate/rotate the XY spatial coordinates to which the orientations are linked... or you can rotate just the orientations (keeping the XY positions), but then the relationship of orientations to the spatial coordinates and structures are divorced from the relationship they had when you imported the data initially.

**Reply by zmichels** [reply]
Can you share/explain a bit more about your goal or the outcome you are trying to achieve? Maybe more info could help.

**Reply by sarialzerey** [reply]
sure, I have 2 grains touching each other, one is on the right and the other is on the left. I want to see what is the orientation of each grains in the mating interface. 
![Capture](https://github.com/mtex-toolbox/mtex/assets/141929646/8748f5a4-f5dc-4f84-adae-756b3085175c)

**Reply by zmichels** [reply]
Ok... in what reference frame do you want to consider or visualize the orientations? (Presumably different from the map where you are already showing them).

Also... the "interface" is perhaps tricky because you only have a mapped trace of a plane, and we don't know for certain the projection of that plane beyond the section surface (unless you do by some other means).

Maybe one avenue to explore is considering any significance of the direction of the grain boundary trace relative to the orientation in each grain. To do that, I would compute the boundary trace in each crystal and then plot both as points in the crystal reference frame (an inverse pole figure).

**Reply by sarialzerey** [reply]
Yeah I agree, the information I would get is limited just to the interface

**Reply by zmichels** [reply]
But importantly not the actual interface (which is a plane)... just the trace (line intersection) of it in the section plane.

**Reply by sarialzerey** [reply]
exactly, let me go off the topic. I'm not a programmer but I need to be better with MTEX. What do I do? I still feel I have limited experience even though I hardly do most of what I want by combining the commands together.

**Reply by zmichels** [reply]
Heres some example code of what I was describing doing. The first part just generates some random versions that you would supplement with the data you have.

cs = loadCIF('quartz');

o1 = orientation.rand(1,cs);
o2 = orientation.rand(1,cs);

BoundaryTrace = vector3d.rand(1);


l1 = o1.\BoundaryTrace;
l2 = o2.\BoundaryTrace;

figure,
plot(symmetrise([l1, l2]), 'antipodal','fundamentalRegion')

**Reply by zmichels** [reply]
Extracting a specific grain boundary trace will require identifying (maybe subsetting) the grains from your larger grainset.

let's say you get grains g1, and g2 isolated, and they have the mineral name, I'll substitute with a variable named "phase"

% grains of interest
goi = [g1 g2];

% Boundary of interest
boi = goi.boundary(phase, phase);

% Segment directions along the boundary
TraceDirs = boi.direction;


%% Orientations along the boundary

% from one grain (one side of the boundary)
o1s = ebsd('id',boi.ebsdId(:,1)).orientations;

% from the other grain (other side of the boundary)
o2s = ebsd('id',boi.ebsdId(:,2)).orientations;


%% Boundary trace directions in crystal reference frame

traceXtal1 = o1s.\TraceDirs;
traceXtal2 = o2s.\TraceDirs;

%% visualize them
figure,
plot(symmetrise(traceXtal1),'antipodal','fundamentalRegion')
hold on
plot(symmetrise(traceXtal2),'antipodal','fundamentalRegion')

**Reply by zmichels** [reply]
Sidenote: That code will use the direction of each *segment* of the larger boundary, and it will use the orientations of the crystal on either side of that segment. So... unless your boundary has only one segment/direction (few do), the code will generate multiple pairs [even though only two grains].

**Reply by sarialzerey** [reply]
Thank you so much! this really helped.
